docs/developers/README.md

```diff
@@ -627,4 +627,37 @@ cat /tekton/steps/step-unnamed-<step-index>/exitCode
 ```
 
 Or, you can access the step metadata directory via symlink, for example, use `cat /tekton/steps/0/exitCode` for the
-first step in a task.
\ No newline at end of file
+first step in a task.
+
+## Experimental Remote Resolution Support
+
+In
+[TEP-0060](https://github.com/tektoncd/community/blob/main/teps/0060-remote-resource-resolution.md)
+we are exploring ways for the Tekton Pipelines controller to offload resolution
+of `taskRefs` and `pipelineRefs` to external systems. The idea is to allow for
+resolution of these resources from locations that aren't supported directly by
+Pipelines, for example pulling in `Tasks` and `Pipelines` from `git` repos.
+
+To support this functionality Pipelines has two experimental flags that allow a
+developer to disable resolution in the TaskRun and PipelineRun reconcilers and
+to enable a new pair of reconcilers dedicated to resolving `taskRefs` and
+`pipelineRefs`.
+
+To disable remote resolution in the TaskRun and PipelineRun reconcilers, set
+the
+[experimental-disable-ref-resolution](../../config/config-feature-flags.yaml)
+feature flag to `"true"`.
+
+To enable the experimental resolution reconcilers add
+`-experimental-enable-resolution-reconcilers` as an additional `arg` passed to
+the `tekton-pipelines-controller` in its [Deployment
+YAML](../../config/controller.yaml).
```

Comment:
this is probably a silly question but im wondering where the line is b/w command line args and configuration we configure via config maps - could it make sense that both of these options are configurable via config map?

(it seems like [we're mostly just using args for images](https://github.com/tektoncd/pipeline/blob/8e1006971158a23750388220cfe6e16aa69f1e21/config/controller.yaml#L63-L80) - not really 100% sure why those wouldn't be configurable via config map also)

Reply:
oh the code is right above - also threads-per-controller, disable-ha, namespace - maybe the idea is these CAN'T change at runtime? (unless they can)

Reply:
I would say yes, can't change at runtime, and need a very very explicit change from the user

Reply:
Two considerations:

1. There's only one spot in our codebase where we can "spin up" reconcilers. That's in [`cmd/controller/main.go`](https://github.com/tektoncd/pipeline/blob/8e1006971158a23750388220cfe6e16aa69f1e21/cmd/controller/main.go#L110-L113). At this early stage in the program's startup we don't have access to configmap values yet.
2. The reconcile loops (`ReconcileKind()`) of our reconcilers don't have direct access to the command line flags passed to the application. We have no existing avenue to pass those flags from the controller's `main.go` down to the reconcilers.

I'm happy to create a way for passing the values from CLI args down to reconcilers - is that what we're discussing here?  I don't think we can spin up / tear down the resolution reconcilers via runtime configmap changes tho.

Reply:
I realise I didn't really address the original point - where the line is b/w command line args and configuration we configure via config maps.  I didn't approach this with a clear line in mind. It was just easiest to plumb the configuration through from both of these sides to their respective points in the application. Open to changes.

Reply:
no problem at all, thanks for the detailed explanation @sbwsg  - it sounds like the line (at least for now) is configuration that impacts how the reconcile loops are started maybe? config for that needs to be passed via command line args, and the rest can be config map

so long story short sounds like you made the right choice to configure this via command line arg

my only remaining question is if we want both options (both the command line arg and the config map param) - i think im not focusing on the most important parts of this proposal so feel free to just make a choice and move on, but my thought is we could initially just have one way of configuring this (maybe the command line option) which both enables the experimental reconciler and disables the current resolution (esp since if this becomes our long term solution im guessing we'd rename or remove `experimental-disable-ref-resolution` anyway) - but also ignore me if that's just more work to make happen than its worth

Reply:
nevermind, my bad, i just thought this through and realized of course you want 2 flags because you want to be able to:

- Turn off resolution and use the new reconcilers to resolve
- Turn off resolution but use some different reconciler to resolve (e.g. git)

👍 👍 👍  to your approach